integration/benchmark/load/evm_batch_load.go

```diff
@@ -0,0 +1,444 @@
+	return EVMBatchTransferLoadType
+}
+
+func (l *EVMBatchTransferLoad) Setup(log zerolog.Logger, lc LoadContext) error {
+
+	// create a shared bridged account
+	err := sendSimpleTransaction(
+		log,
+		lc,
+		func(
+			log zerolog.Logger,
+			lc LoadContext,
+			acc *account.FlowAccount,
+		) (*flowsdk.Transaction, error) {
+			l.bridgedAcountAddress = acc.Address
+
+			sc := systemcontracts.SystemContractsForChain(lc.ChainID)
+
+			contractName := "BridgedAccountContract"
+
+			contract := fmt.Sprintf(`
+				import EVM from %s
+				import FlowToken from %s
+
+				access(all) contract BridgedAccountContract {
+					access(self) var acc: @EVM.CadenceOwnedAccount
+
+					access(all)
+					fun address() : EVM.EVMAddress {
+						return self.acc.address()
+					}
+
+					access(all)
+					fun call(
+						to: EVM.EVMAddress,
+						data: [UInt8],
+						gasLimit: UInt64,
+						value: EVM.Balance
+					): EVM.Result {
+						return self.acc.call(
+							to: to,
+							data: data,
+							gasLimit: gasLimit,
+							value: value
+						)
+					}
+
+					access(all)
+					fun deposit(from: @FlowToken.Vault) {
+						self.acc.deposit(from: <-from)
+					}
+
+					init() {
+						self.acc <- EVM.createCadenceOwnedAccount()
+					}
+				}
+			`,
+				sc.EVMContract.Address.HexWithPrefix(),
+				sc.FlowToken.Address.HexWithPrefix())
+
+			tx := flowsdk.NewTransaction().
+				SetScript(blueprints.DeployContractTransactionTemplate)
+
+			err := tx.AddArgument(cadence.String(contractName))
+			if err != nil {
+				return nil, err
+			}
+			err = tx.AddArgument(cadence.String(contract))
+			if err != nil {
+				return nil, err
+			}
+
+			return tx, nil
+		})
+	if err != nil {
+		return fmt.Errorf("error creating shared bridged account: %w", err)
+	}
+
+	// create some EOA ahead of time to get a better result for the benchmark
+	createEOA := l.PreCreateEOAAccounts
+
+	g, ctx := errgroup.WithContext(context.Background())
+	g.SetLimit(lc.Proposer.NumKeys())
+
+	progress := util.LogProgress(l.log,
+		util.DefaultLogProgressConfig(
+			"creating and funding EOC accounts",
+			createEOA,
+		))
+
+	l.log.Info().
+		Int("number_of_accounts", createEOA).
+		Int("number_of_keys", lc.Proposer.NumKeys()).
+		Msg("creating and funding EOC accounts")
+
+	for i := 0; i < createEOA; i += 1 {
+		i := i
+		g.Go(func() error {
+			select {
+			case <-ctx.Done():
+				return nil
+			default:
+			}
+			defer func() { progress(1) }()
+
+			eoa, err := l.setupTransaction(log, lc)
+			if err != nil {
+				return err
+			}
+
+			if err != nil {
+				l.log.
+					Err(err).
+					Int("index", i).
+					Msg("error creating EOA accounts")
+				return err
+			}
+
+			l.creations.Add(1)
+
+			select {
+			case l.eoaChan <- eoa:
+			default:
+			}
+
+			return nil
+		})
+	}
+	err = g.Wait()
+	if err != nil {
+		return fmt.Errorf("error creating EOC accounts: %w", err)
+	}
+	return nil
+}
+
+func (l *EVMBatchTransferLoad) Load(log zerolog.Logger, lc LoadContext) error {
+	select {
```

Comment:
is there a way to gracefully shutdown?

Reply:
The load function is called by workers. Those workers will ultimately be stopped and garbage collected, at which point this will be too.